--- driver.c.orig
+++ driver.c
@@ -165,12 +165,10 @@ spawnphase(struct stageinfo *phase, int *fd, char *input, char *output, bool las
 			goto err1;
 		}
 		if (fcntl(pipefd[0], F_SETFD, FD_CLOEXEC) < 0) {
-			ret = errno;
-			goto err2;
+			warn("fcntl(..., F_SETFD, FD_CLOEXEC):");
 		}
 		if (fcntl(pipefd[1], F_SETFD, FD_CLOEXEC) < 0) {
-			ret = errno;
-			goto err2;
+			warn("fcntl(..., F_SETFD, FD_CLOEXEC):");
 		}
 		ret = posix_spawn_file_actions_adddup2(&actions, pipefd[1], 1);
 		if (ret)
